### DISCUSSION TITLE
Is `populate` cached?

### DISCUSSION THREAD

**Discussion by corrm** [mikro-orm/mikro-orm#6025] [Q&A] [answered]
```ts
public async levelUp(userCharacter: UserCharacter): Promise<boolean> {
        await this.em.populate(userCharacter, ["character"]);

        if (userCharacter.currentLevel >= userCharacter.character.maxLevel.level) {
            return false;
        }

        userCharacter.currentLevel += 1;
        await this.em.persistAndFlush(userCharacter);

        return true;
    }
```

```
    @ManyToOne(() => Character)
    @Index()
    public readonly character: Rel<Character>;
```

if `userCharacter.character` already loaded, did `populate` do a DB query?
if yes is there any better approach to check if data loaded before do DB query?
can check `if (userCharacter.character === undefined)` but if there is any built-in method would be great.

**Comment by B4nan** [comment, marked as answer]
> if userCharacter.character already loaded, did populate do a DB query?

Nope, it will skip loaded relations unless you use `refresh: true`.

> if yes is there any better approach to check if data loaded before do DB query?

You can use `wrap(userCharacter.character).isInitialized()`.

> can check if (userCharacter.character === undefined) but if there is any built-in method would be great.

Not really, this would only check if the relation is missing because of partial loading, or simply not set at all. If it's not populated, the value is still there, it's just an entity reference with the FK value present.

**Reply by corrm** [reply]
One more question to make all clear:
is both code will not do DB query (so I could just use `populate` instead of `wrap.isInitialized` check then call `populate`)
```ts
if (!wrap(userCharacter.character).isInitialized()) {
    await this.em.populate(userCharacter, ["character.maxLevel"]);
}

// without `refresh: true`
await this.em.populate(userCharacter, ["character.maxLevel"]);
```

**Reply by B4nan** [reply]
Yeah, that should be safe.

**Reply by corrm** [reply]
would be nice to add this tip to `populate` function docs.

_Thanks for your time_
